src/controllers/authController.js

```diff
@@ -0,0 +1,239 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const userService = require('../services/userService.js');
+const jwtService = require('../services/jwtService.js');
+const tokenService = require('../services/tokenService.js');
+const emailService = require('../services/emailService.js');
+const { ApiError } = require('../exceptions/ApiError.js');
+const { User } = require('../models/User.js');
+
+const REFRESH_TOKEN_MAX_DAYS = 30 * 24 * 60 * 60 * 1000; // 30 days
+
+async function register(req, res) {
+  const { name, email, password } = req.body;
+
+  const emailError = userService.validateEmail(email);
+
+  if (emailError) {
+    throw ApiError.BadRequest('Validation error', { email: emailError });
+  }
+
+  const passwordError = userService.validatePassword(password);
+
+  if (passwordError) {
+    throw ApiError.BadRequest('Validation error', { password: passwordError });
+  }
+
+  await userService.register({
+    name,
+    email,
+    password,
+  });
+
+  res.send({ message: 'User created' });
```

Comment:
For better user experience and debugging, consider returning the user data after creation.